homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,207 @@
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug(f"create sensor from config entry {config_entry.data}")
+    session = async_get_clientsession(hass)
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    try:
+        district = config_entry.data[ATTR_COUNTY]
+        sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in SENSORS
+        ]
+        async_add_entities(sensors, update_before_add=True)
+    except KeyError:
+        _LOGGER.error("Could not determine %s from config!", ATTR_COUNTY)
+
+
+class RKICovidNumbersSensor(CoordinatorEntity):
+    """Representation of a sensor."""
+
+    name = None
+    unique_id = None
+
+    def __init__(
+        self,
+        coordinator: update_coordinator.DataUpdateCoordinator,
+        district: Dict[str, str],
+        info_type: str,
+    ):
+        """Initialize a new sensor."""
+        _LOGGER.debug(f"initialize {info_type} sensor for {district}")
+        super().__init__(coordinator)
+
+        data = coordinator.data[district]
+
+        if data.county:
+            self.name = f"{data.county} {info_type}"
+        else:
+            self.name = f"{data.name} {info_type}"
+        self.unique_id = f"{district}-{info_type}"
+        self.district = district
+        self.info_type = info_type
+        self.updated = datetime.now()
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            self.coordinator.last_update_success
+            and self.district in self.coordinator.data
+        )
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return current state."""
+        try:
+            return getattr(self.coordinator.data[self.district], self.info_type)
+        except AttributeError:
+            return None
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return SENSORS[self.info_type]
+
+    @property
+    def state_class(self):
+        """Opt-in for long-term statistics."""
+        return STATE_CLASS_MEASUREMENT
+
+    @property
+    def unit_of_measurement(self):
```

Comment:
Don't use `unit of measure`, use `native_unit_of_measure`

Reply:
I thought the `native_unit_of_measurement` should only be used if the unit will be modified by the SensorEntity
https://developers.home-assistant.io/blog/2021/08/12/sensor_temperature_conversion?_highlight=unit_of_measurement#native_unit_of_measurement

Reply:
No, should always be used
https://github.com/home-assistant/core/blob/dev/homeassistant/components/sensor/__init__.py#L261

Reply:
This is also recommended in the official sample.
https://github.com/home-assistant/example-custom-config/blob/master/custom_components/example_sensor/sensor.py#L39

Reply:
done